src/api/user.js

```diff
@@ -55,5 +95,50 @@ module.exports = {
         }
       }
     })
+  },
+
+  updatePassword (userId, hashNewPassword, callback, error) {
+    User.updateOne({
+      '_id': userId
+    }, {
+      $set: {
+        'password': hashNewPassword
+      }
+    }, (err, result) => {
+      if (err) {
+        error(new Error('No password changed'))
+      } else {
+        callback(result)
```

Comment:
What is returning here? Usually `update` functions returns `true` or `false` indicating whether the operation is succeeded or not. I would suggest you use the `result.n_updated` (make sure you go to the documentation) and check if there's exactly one db entry got updated.